kas-fleet-manager/deploy-kas-fleet-manager.sh

```diff
@@ -141,35 +88,71 @@ deploy_kasfleetmanager() {
 
+  create_kasfleetmanager_service_account
+  create_kasfleetmanager_pull_credentials
+
   echo "Deploying KAS Fleet Manager K8s Secrets..."
   ${OC} process -f ${KAS_FLEET_MANAGER_CODE_DIR}/templates/secrets-template.yml \
-    -p OCM_SERVICE_CLIENT_ID="dummyclient" \
-    -p OCM_SERVICE_CLIENT_SECRET="dummysecret" \
+    -p OCM_SERVICE_CLIENT_ID="" \
+    -p OCM_SERVICE_CLIENT_SECRET="" \
+    -p OCM_SERVICE_TOKEN="${OCM_SERVICE_TOKEN}" \
     -p OBSERVABILITY_CONFIG_ACCESS_TOKEN="${OBSERVABILITY_CONFIG_ACCESS_TOKEN}" \
     -p MAS_SSO_CLIENT_ID="${MAS_SSO_CLIENT_ID}" \
     -p MAS_SSO_CLIENT_SECRET="${MAS_SSO_CLIENT_SECRET}" \
+    -p OSD_IDP_MAS_SSO_CLIENT_ID="${MAS_SSO_CLIENT_ID}" \
+    -p OSD_IDP_MAS_SSO_CLIENT_SECRET="${MAS_SSO_CLIENT_SECRET}" \
     -p MAS_SSO_CRT="${MAS_SSO_CRT}" \
     -p KAFKA_TLS_CERT="${KAFKA_TLS_CERT}" \
     -p KAFKA_TLS_KEY="${KAFKA_TLS_KEY}" \
     -p DATABASE_HOST="$(${KUBECTL} get service/kas-fleet-manager-db -o jsonpath="{.spec.clusterIP}")" \
+    -p KUBE_CONFIG="$(${OC} config view --minify --raw | base64 -w0)" \
+    -p IMAGE_PULL_DOCKER_CONFIG=$(${OC} get secret ${KAS_FLEET_MANAGER_IMAGE_PULL_SECRET_NAME} -n ${KAS_FLEET_MANAGER_NAMESPACE} -o jsonpath="{.data.\.dockerconfigjson}") \
     | ${OC} apply -f - -n ${KAS_FLEET_MANAGER_NAMESPACE}
 
   echo "Deploying KAS Fleet Manager Envoy ConfigMap..."
   ${OC} apply -f ${KAS_FLEET_MANAGER_CODE_DIR}/templates/envoy-config-configmap.yml -n ${KAS_FLEET_MANAGER_NAMESPACE}
 
-  create_kasfleetmanager_service_account
-  create_kasfleetmanager_pull_credentials
-
   echo "Deploying KAS Fleet Manager..."
   OCM_ENV="development"
 
+  SERVICE_PARAMS=${DIR_NAME}/kas-fleet-manager-params.env
+
+  if [ -n "${OCM_SERVICE_TOKEN}" ] ; then
+      PROVIDER_TYPE="ocm"
+      CLUSTER_STATUS="cluster_provisioned"
+      ENABLE_READY_DATA_PLANE_CLUSTERS_RECONCILE="true"
+
+      if [ -n "${STRIMZI_OPERATOR_SUBSCRIPTION_CONFIG}" ] ; then
+          echo "WARN: Strimzi operator subscription config will not be used with the 'ocm' cluster provider type"
+      fi
+
+      if [ -n "${KAS_FLEETSHARD_OPERATOR_SUBSCRIPTION_CONFIG}" ] ; then
+          echo "WARN: Fleetshard operator subscription config will not be used with the 'ocm' cluster provider type"
+      fi
+  else
+      PROVIDER_TYPE="standalone"
+      CLUSTER_STATUS="ready"
+      ENABLE_READY_DATA_PLANE_CLUSTERS_RECONCILE="false"
+  fi
+
+  > ${SERVICE_PARAMS}
+
+  if [ -n "${KAS_FLEET_MANAGER_SERVICE_TEMPLATE_PARAMS:-}" ] ; then
+      echo "${KAS_FLEET_MANAGER_SERVICE_TEMPLATE_PARAMS}" >> ${SERVICE_PARAMS}
+  fi
+
+  if [ -n "${SUPPORTED_INSTANCE_TYPES:-}" ] ; then
+      echo "SUPPORTED_INSTANCE_TYPES='${SUPPORTED_INSTANCE_TYPES}'" >> ${SERVICE_PARAMS}
+  fi
+
   ${OC} process -f ${KAS_FLEET_MANAGER_CODE_DIR}/templates/service-template.yml \
+    --param-file=${SERVICE_PARAMS} \
     -p ENVIRONMENT="${OCM_ENV}" \
-    -p OCM_URL="https://nonexistingdummyhosttest.com" \
+    -p OCM_URL="https://api.stage.openshift.com" \
```

Comment:
```suggestion
    -p OCM_URL="https://api.stage.openshift.com" \
    -p AMS_URL="https://api.stage.openshift.com" \ 
```
let's set also the AMS_URL. It is useful for admin endpoints. It only works when the ocm service account is set.

Reply:
I can add this conditionally using the `param-file`. Should `OCM_URL` also be dynamic depending on whether we're using the ocm provider?

Reply:
Yes. The OCM_URL can also be dynamic depending on whether we are using the ocm provider. When we are not using the ocm provider, let's enable OCM_MOCK.

Reply:
Added dynamic configs for OCM and AMS URL vars when using the ocm provider.